### ISSUE TITLE
Infrastructure for ancient samples

### ISSUE THREAD

**Issue by jeromekelleher** [popgensims/stdpopsim#166]
Some models require ancient samples (see #146 and discussion in #140). Add infrastructure to specify a ``sampling_time``. This can replace the existing ``allow_samples`` attribute, doing something like 

```python
class Population(object):
    """
    Class recording metadata representing a population in a simulation.
    """    
    def __init__(self, name, description,sampling_time=None):
        self.name = name
        self.description = description
        self.sampling_time = sampling_time
    @property
    def allow_samples(self):
          return self.sampling_time is not None
```

So, this should give us the ability to specify sampling times or not. It'll need a little updating of existing model code, but that should be OK.

**Comment by ndukler** [comment]
I'm coding this up over the next few days

**Comment by jeromekelleher** [comment]
Closed in #177.
